LibOS/shim/src/fs/shim_fs.c

```diff
@@ -328,16 +404,57 @@ int mount_fs (const char * type, const char * uri, const char * mount_point)
     if ((ret = fs->fs_ops->mount(uri, mount_point, &mount_data)) < 0)
         goto out;
 
+
     int uri_len = uri ? strlen(uri) : 0;
-    qstrsetstr(&mount->path, mount_point, strlen(mount_point));
+    qstrsetstr(&mount->path, mount_point, mount_point_len);
     qstrsetstr(&mount->uri, uri, uri_len);
     memcpy(mount->type, fs->name, sizeof(fs->name));
     mount->fs_ops    = fs->fs_ops;
     mount->d_ops     = fs->d_ops;
     mount->data      = mount_data;
+
+    /* Get the negative dentry from the cache, if one exists */
+    struct shim_dentry * dent, *dent2;
+    /* Special case the root */
+    if (mount_point_len == 1 && mount_point[0] == '/')
+        dent = dentry_root;
+    else {
+        dent = __lookup_dcache(parent, last,
+                               last_len,
+                               NULL, 0, NULL);
+
+        if(!dent) {
+            dent = get_new_dentry(mount, parent, last, last_len, NULL);
+            get_dentry(dent);
+        }
+    }
+    
+    assert(dent == dentry_root || !(dent->state & DENTRY_VALID));    
+
+    // We need to fix up the relative path to this mount, but only for
+    // directories.  
+    qstrsetstr(&dent->rel_path, "", 0);
     mount->path.hash = dent->rel_path.hash;
 
+    /*Now go ahead and do a lookup so the dentry is valid */
+    if ((ret = __path_lookupat(dentry_root, mount_point, 0, &dent2, 0,
+                               parent ? parent->fs : mount, make_ancestor)) < 0) 
+        goto out;
+
+    assert(dent == dent2);
+        
```

Comment:
dent2 is never released.

Reply:
I think you are right.  Let me test this change.